packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/MethodCallHandlerImpl.java

```diff
@@ -49,7 +52,18 @@ public void onMethodCall(MethodCall call, MethodChannel.Result result) {
       case "setShortcutItems":
         List<Map<String, String>> serializedShortcuts = call.arguments();
         List<ShortcutInfo> shortcuts = deserializeShortcuts(serializedShortcuts);
-        shortcutManager.setDynamicShortcuts(shortcuts);
+        ThreadPoolExecutor executor =
+            new ThreadPoolExecutor(0, 1, 1, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>());
+
+        executor.execute(
+            new Runnable() {
+              @Override
+              public void run() {
+                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.N_MR1) {
```

Comment:
Shouldn't this check be around all of the new code? Presumably we don't want to start a thread and send a task to it just to no-op on older versions of Android.